### PR TITLE
Add alerts and dashboards

### PR DESCRIPTION
**Pull request by kakkoyun**
cc @squat

### CODE REVIEW

**Comment by squat** [comment]
Can you re-commit this without adding the jsonnet/vendor dir? normally we expect that to be generated by jb with the lockfile, so there's no need to commit it to the repo

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+SRC = $(shell find . -type f -name '*.go' -not -path './vendor/*')
+
+all: fmt thanos-receive-controller
+
+thanos-receive-controller: main.go
+	go build -v
+
+.PHONY: generate
+generate: alerts.yaml rules.yaml dashboards
+
```
Comment (squat) [inline Makefile]: there seems to be no makefile rule for these targets

File: jsonnet/Makefile [hunk 2]
```diff
@@ -0,0 +1,58 @@
+EXAMPLES := ../examples
```
Comment (squat) [inline jsonnet/Makefile]: let's merge this Makefile with the one at the root so we only have one

File: jsonnet/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+vendor
```
Comment (squat) [inline jsonnet/.gitignore]: it should be sufficient to have the entry in the .gitignore at the root, no?

Comment (kakkoyun) [reply jsonnet/.gitignore]: Definitely, I've forgotten to clean things up

**Comment by kakkoyun** [comment]
cc @metalmatze

**Comment by brancz** [comment]
Why are we committing a lockfile here at all?

**Comment by squat** [comment]
ah, frederic means that since this repo doesnt actually commit any generated manifests, i.e. it is only used as a library that is imported by other jsonnet projects, it doesn't need to lock and dependencies.

**Comment by kakkoyun** [comment]
@brancz for no particular reason, I didn't know we shouldn't. I'll remove it.

**Comment by kakkoyun** [comment]
@squat Got that now 👍 Thanks for the explanation :)

**Comment by brancz** [comment]
lgtm, leaving last call to @squat @metalmatze

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		-v "$(PWD):/go/src/github.com/observatorium/thanos-receive-controller:Z" \
+		-w "/go/src/github.com/observatorium/thanos-receive-controller" \
+		-e USER=deadbeef \
+		-e GO111MODULE=on \
+		quay.io/coreos/jsonnet-ci
+
+all: generate fmt thanos-receive-controller
+
+thanos-receive-controller: main.go
+	go build -v
```
Comment (squat) [inline Makefile]: We should make sure this uses the same flags and environment variables as we have today in the build section of the drone.yml. Then, we should change the command in drone.yml to instead run `make thanos-receive-controller`

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	jsonnet jsonnet/thanos-receive-controller-mixin/rules.jsonnet | gojsontoyaml > $@
+
+.PHONY: vendor
+vendor: go.mod go.sum jsonnet/jsonnetfile.json jsonnet/jsonnetfile.lock.json
+	rm -rf jsonnet/vendor
+	cd jsonnet && jb install
+	go mod vendor
+
+.PHONY: fmt
+fmt:
```
Comment (squat) [inline Makefile]: drone.yml should invoke this rather than it's own script in the fmt section.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+	@test -z $(shell fmt_res=$(gofmt -d -s ${SRC})) || printf '\nGofmt found style issues. Please check the reported issues\nand fix them if necessary before submitting the code for review:\n\n%s' "$(fmt_res)"
+	echo ${JSONNET_SRC} | xargs -n 1 -- $(JSONNET_FMT) -i
+
+.PHONY: lint
+lint: fmt ${ALERTS} ${RULES}
```
Comment (squat) [inline Makefile]: we should add this to CI

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+	-golangci-lint run -v --enable-all
+	promtool check rules ${ALERTS} ${RULES}
+
+.PHONY: test ${ALERTS} ${RULES}
+test:
```
Comment (squat) [inline Makefile]: ensure `make test` is invoked by ci in the test section. we'll need to use an docker image that includes promtool.

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+		-e GO111MODULE=on \
+		quay.io/coreos/jsonnet-ci
+
+all: generate fmt thanos-receive-controller
+
+thanos-receive-controller: main.go
+	go build -v
+
+.PHONY: generate
+generate: ${ALERTS} ${RULES} ${DASHBOARDS}
```
Comment (squat) [inline Makefile]: Our CI should have a `generate` step that runs `make generate` and then ensures that `git diff --exit-code` exits clean.

**Comment by kakkoyun** [comment]
@squat I have fixed emphasized make issues. I introduced two new docker images as drone stages
https://github.com/kakkoyun/docker-go-promtool and https://github.com/kakkoyun/docker-go-jsonnet-ci. I can put them some other common registry, I've just put them in my personal account for now.

**Comment by squat** [comment]
To simplify the maintenance of docker containers, I think we should use the `quay.io/coreos/jsonnet-ci` container. We use this in several other projects (prometheus-operator, CMO, telemeter etc). The only thing it is missing is promtool, so we'd have to add that. This way you don't need to build and keep the other two containers up to date. WDYT?

**Comment by kakkoyun** [comment]
@squat Totally makes sense. I'll send a PR to add `promtool`. I guess this is the place https://github.com/coreos/prometheus-operator/blob/master/scripts/jsonnet/Dockerfile

**Comment by brancz** [comment]
We can move forward here now, the image is about to be published.

**Comment by kakkoyun** [comment]
@squat `quay.io/coreos/jsonnet-ci` is updated as we talked https://github.com/coreos/prometheus-operator/pull/2739. It's merged and this PR updated. Ready for your review.

**Comment by brancz** [comment]
lgtm giving @squat last call

**Pull request merged**